lib/optimize/ModuleConcatenationPlugin.js

```diff
@@ -40,50 +44,51 @@ class ModuleConcatenationPlugin {
 				for(const module of modules) {
 					// Only harmony modules are valid for optimization
 					if(!module.meta || !module.meta.harmonyModule) {
+						setBailoutReason(module, "Module is not an ECMAScript module");
 						continue;
 					}
 
 					// Because of variable renaming we can't use modules with eval
 					if(module.meta && module.meta.hasEval) {
-						setBailoutReason(module, "ModuleConcatenation", "eval is used in the module");
+						setBailoutReason(module, "Module uses eval()");
 						continue;
 					}
 
 					relevantModules.push(module);
 
 					// Module must not be the entry points
 					if(module.getChunks().some(chunk => chunk.entryModule === module)) {
-						setBailoutReason(module, "ModuleConcatenation (inner)", "module is an entrypoint");
+						setBailoutReason(module, "Module is an entry point");
 						continue;
 					}
 
 					// Exports must be known (and not dynamic)
 					if(!Array.isArray(module.providedExports)) {
-						setBailoutReason(module, "ModuleConcatenation (inner)", "exports are not known");
+						setBailoutReason(module, "Module exports are unknown");
 						continue;
 					}
 
 					// Using dependency variables is not possible as this wraps the code in a function
 					if(module.variables.length > 0) {
-						setBailoutReason(module, "ModuleConcatenation (inner)", "dependency variables are used (i. e. ProvidePlugin)");
+						setBailoutReason(module, "Module uses injected variables (usually caused by the ProvidePlugin)");
 						continue;
 					}
 
 					// Module must only be used by Harmony Imports
 					const nonHarmonyReasons = module.reasons.filter(reason => !(reason.dependency instanceof HarmonyImportDependency));
 					if(nonHarmonyReasons.length > 0) {
 						const importingModules = new Set(nonHarmonyReasons.map(r => r.module));
-						setBailoutReason(module, "ModuleConcatenation (inner)", (requestShortener) => {
+						setBailoutReason(module, (requestShortener) => {
 							const names = Array.from(importingModules).map(m => m.readableIdentifier(requestShortener)).sort();
-							return `module is used with non-harmony imports from ${names.join(", ")}`;
+							return `Module is referenced from these non-ECMAScript modules: ${names.join(", ")}`;
```

Comment:
This message is not totally correct. You could use `require()` instead a ECMAScript module to trigger this bailout.

This bailout is trigger if you use any other way of declaring a dependency except static `import`. `import()` is not a static import, so it will trigger bailout too.